api-test/internal/esignet/driver.go

```
@@ -0,0 +1,715 @@
+	}
+}
+
+// do performs one request, records it (headers/cookies/body), and returns the
+// response body, status, and Location header. When follow is false, redirects
+// are not followed (so we can read the Location).
+func (s *session) do(label, method, u string, body []byte, contentType string, follow bool) ([]byte, int, string, error) {
+	var rdr io.Reader
+	if body != nil {
+		rdr = bytes.NewReader(body)
+	}
+	req, err := http.NewRequest(method, u, rdr)
```

golangci-lint (2.12.2)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)